src/lib/cht-session.ts

```diff
@@ -51,15 +52,21 @@ export default class ChtSession {
     return this.facilityIds.includes(ADMIN_FACILITY_ID);
   }
 
-  public static async create(authInfo: AuthenticationInfo, username : string, password: string): Promise<ChtSession> {
-    const sessionToken = await ChtSession.createSessionToken(authInfo, username, password);
+  public static async create(authInfo: AuthenticationInfo, username: string, password: string): Promise<ChtSession> {
+    try {
+      const sessionToken = await ChtSession.createSessionToken(authInfo, username, password);
+      if (!sessionToken) {
+        throw new Error(`failed to obtain token for ${username} at ${authInfo.domain}`);
+      }
 
-    if (!sessionToken) {
-      throw new Error(`failed to obtain token for ${username} at ${authInfo.domain}`);
+      const creationDetails = await ChtSession.fetchCreationDetails(authInfo, username, sessionToken);
+      return new ChtSession(creationDetails);
+    } catch (e: any) {
+      if (e.response?.status === 401) {
```

Comment:
Might be nice to have `AuthErrors.assertFromException(e)` for encapsulation?
I suspect the try/catch should be in createSessionToken? But that's up to you